### ISSUE TITLE
Chrome Canary/Firefox Nightly test results

### ISSUE THREAD

**Issue by OrKoN** [puppeteer/puppeteer#12379]
The bot will post failures here if Puppeteer tests fail with the canary build.

**Comment by OrKoN** [comment]
cc @Lightning00Blade the latest failure is caused by the flip of the default for the --headless flag

**Comment by amerj0505-cell** [comment]
P1

**Comment by whimboo** [comment]
@Lightning00Blade  regarding the failure for Firefox builds @lutien is going to create a PR for it.

**Comment by Lightning00Blade** [comment]
FYI @whimboo It seems the following test started failing for Windows with latest nightly versions
Looking at the comments most likely a change between 09/04/2026 and 10/04/2026.
```
BrowserContext
       BrowserContext.overridePermissions
         should grant permission when listed:
```

**Comment by OrKoN** [comment]
@Lightning00Blade this should be fixed with https://bugzilla.mozilla.org/show_bug.cgi?id=2031230

**Comment by whimboo** [comment]
Correct. If you want to re-run today's job you can do given that the new nightly is out now and the tests should no longer fail.

**Comment by whimboo** [comment]
Looks all to be good now. No more failures today for Firefox!

**Comment by whimboo** [comment]
The geolocation test that is timing out today on macOS is a regression and I filed https://bugzilla.mozilla.org/show_bug.cgi?id=2033385.

**Comment by whimboo** [comment]
> The geolocation test that is timing out today on macOS is a regression and I filed https://bugzilla.mozilla.org/show_bug.cgi?id=2033385.

This is now fixed and I'm going to uplift to Firefox beta 151 as well.

**Comment by OrKoN** [comment]
cc @wolfib looks like we have a CDP failure

**Comment by whimboo** [comment]
The test which is regularly failing for Firefox is:

* `[waittask.test] waittask specs protocol timeout \"before all\" hook for \"should error if underlying protocol command times out with raf polling\"`
* `[waittask.test] waittask specs protocol timeout \"after all\" hook for \"should error if underlying protocol command times out with raf polling\"`

So we already increased the timeout from 3s to 5s via https://github.com/puppeteer/puppeteer/pull/14649. While this made the failure not permanent anymore we have it flaky now. So it looks like it's still not enough and Firefox takes longer to launch on Windows in your CI environment.

@OrKoN shall we consider bumping up the value even more or try the alternative solution:

1. Add a `setTimeout()` method to `BidiConnection` so the timeout can be changed after construction
2. In `mocha-utils.ts`, launch the browser with the computed (long) `protocolTimeout`, then apply the test's short `protocolTimeout` after the browser is up.

But not sure if that defeats the purpose of this test.

**Comment by whimboo** [comment]
> [@OrKoN](https://github.com/OrKoN) shall we consider bumping up the value even more or try the alternative solution:
> 
>     1. Add a `setTimeout()` method to `BidiConnection` so the timeout can be changed after construction
> 
>     2. In `mocha-utils.ts`, launch the browser with the computed (long) `protocolTimeout`, then apply the test's short `protocolTimeout` after the browser is up.
> 
> 
> But not sure if that defeats the purpose of this test.

Has something changed with the Windows workers recently? All the last days I was not able to see these tests failing in Github actions.

**Comment by OrKoN** [comment]
> > [@OrKoN](https://github.com/OrKoN) shall we consider bumping up the value even more or try the alternative solution:
> > ```
> > 1. Add a `setTimeout()` method to `BidiConnection` so the timeout can be changed after construction
> > 
> > 2. In `mocha-utils.ts`, launch the browser with the computed (long) `protocolTimeout`, then apply the test's short `protocolTimeout` after the browser is up.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But not sure if that defeats the purpose of this test.
> 
> Has something changed with the Windows workers recently? All the last days I was not able to see these tests failing in Github actions.

Note that I know of. Either something changed in Firefox or with the infrastructure. I think we had no changes in Puppeteer.

**Comment by whimboo** [comment]
> Note that I know of. Either something changed in Firefox or with the infrastructure. I think we had no changes in Puppeteer.

You need to start talking about it to let it fail again. So it might be lesser present maybe these days. @OrKoN what would you say about https://github.com/puppeteer/puppeteer/issues/12379#issuecomment-4853188093? Does that sound ok?

**Comment by OrKoN** [comment]
> > Note that I know of. Either something changed in Firefox or with the infrastructure. I think we had no changes in Puppeteer.
> 
> You need to start talking about it to let it fail again. So it might be lesser present maybe these days. [@OrKoN](https://github.com/OrKoN) what would you say about [#12379 (comment)](https://github.com/puppeteer/puppeteer/issues/12379#issuecomment-4853188093)? Does that sound ok?

I think I would prefer to increase the timeout and not allow changing the protocol timeout at runtime.

**Comment by whimboo** [comment]
@OrKoN recently I saw the [following failure twice for Firefox tests](https://github.com/puppeteer/puppeteer/actions/runs/32942111504/job/98095044227#step:7:18): 

```
npm error Lifecycle script `build` failed with error:
npm error code 3221226505
npm error path D:\a\puppeteer\puppeteer\test
npm error workspace @puppeteer-test/test@latest
npm error location D:\a\puppeteer\puppeteer\test
npm error command failed
npm error command C:\Windows\system32\cmd.exe /d /s /c wireit
Error: Process completed with exit code 1.
```
